crates/cli_bin/tests/format.rs

```diff
@@ -0,0 +1,60 @@
+use crate::common::get_test_cmd;
+use anyhow::Result;
+use common::get_fixture;
+use insta::assert_yaml_snapshot;
+
+mod common;
+
+#[test]
+fn format_patterns() -> Result<()> {
+    let (_temp_dir, grit_dir) = get_fixture("unformatted_patterns", true)?;
+
+    let mut cmd = get_test_cmd()?;
+    cmd.arg("format").current_dir(grit_dir.clone());
+    let output = cmd.output()?;
+
+    println!("stderr: {}", String::from_utf8(output.stderr.clone())?);
+    println!("stdout: {}", String::from_utf8(output.stdout.clone())?);
+
+    assert!(
+        output.status.success(),
+        "Command didn't finish successfully"
+    );
+    assert_yaml_snapshot!(String::from_utf8(output.stdout)?);
```

Comment:
Instead of snapshotting the test output, leading to [these problems](https://github.com/getgrit/gritql/pull/595/files#r1902697985), snapshot the final state of the file (after formatting) + assert on test output containing known messages.

Reply:
> leading to [these problems](https://github.com/getgrit/gritql/pull/595/files#r1902697985)

what problems? the link just opens pull request file changes.

Reply:
done